### PR TITLE
Update configuration for Pydantic V2.0

### PR DESCRIPTION
**Pull request by oldfielj-ansto**

### CODE REVIEW

**Comment by oldfielj-ansto** [comment]
@marcus-oscarsson Might be a PR for you.

Updating as all dependencies of MXCuBE must support Pydantic V2.0 before we can migrate MXCuBE Web/Core over.

**Comment by beteva** [comment]
Hi @oldfielj-ansto, do you really need the *.pyc to be included in the PR?

**Comment by oldfielj-ansto** [comment]
> Hi @oldfielj-ansto, do you really need the *.pyc to be included in the PR?

Ah right, so those changes in the diff are to remove the `*.pyc` files, for some reason there wasn't a `.gitignore` file added for this repository and these were previously committed.

File: pyproject.toml [hunk 1]
```diff
@@ -56,6 +57,22 @@ exclude = '''
 )/
 '''
 
+[tool.pyright]
```
Comment (marcus-oscarsson) [inline pyproject.toml:60]: Nice :) I have not started to use  `pyright` or `ruff` yet but its a nice addition !

**Comment by marcus-oscarsson** [comment]
Nice thanks !

**Pull request merged**